examples/llm/src/models/param_init_fns.py

```diff
@@ -293,9 +293,99 @@ def xavier_normal_param_init_fn_(module, cfg):
     generic_param_init_fn_(module, cfg, xavier_normal_)
 
 
+def deepnorm_init_fn_(module, cfg):
+    # from DeepNet by Wang et al 2022 https://arxiv.org/abs/2203.00555
+    if cfg.get('verbose') and cfg.get('verbose') > 1:
+        warnings.warn(
+            f'If model has bias parameters they are initialized to 0.')
+
+    beta = (8 * cfg.n_layers)**-0.25
+    fvo_init_fn_ = partial(nn.init.xavier_normal_, gain=beta)
+    qk_init_fn_ = partial(nn.init.xavier_normal_, gain=1.)
+
+    if isinstance(module, nn.Linear):
+        # Linear
+        if hasattr(module, '_fused'):
+            fused_init_helper_(module, fvo_init_fn_)
+        else:
+            fvo_init_fn_(module.weight)
```

Comment:
Note:
Later on when we init
```
    elif isinstance(module, nn.MultiheadAttention):
        # torch's MultiheadAttention
```
this is only initializing `torch.nn.MultiheadAttention`.
other layers implementations just use linear for Wqkv (which is a fused layer) and therefore `fused_init_helper_` will init them using `fvo_init_fn_`, when I think you want to init them using Wq and Wk using `qk_init_fn_`

Reply:
I don't know how to change `fused_init_helper_` in a forward compatible way. I currently just want to make it `fused_init_helper_(module, qk_init, fvo_init)`, which I know makes it less general... but I am not sure what type of generality we will want in the future?

Reply:
Since its a super custom init_fn_, just dont use the fused init helper and do it by hand